scalableminds/webknossos
# Segments lists

This PR:
- changes the old "Meshes" tab to a "Segments" tab
- renders "Segments" in the new tab. The list of segments is not guaranteed to be complete or up to date. The list "grows" while creating an annotation or browsing a dataset. For example, selecting an existing segment or drawing with a new segment id will both ensure that the segment is listed. Outdated segments are not handled at all, right now. In a follow-up, we can validate existing segments ad-hoc and show an "outdated" warning etc.
- When loading a mesh, the mesh will be added as child to the segment (at least visually).
- The mesh UI (i.e., the header in the segments tab) was reworked so that the user can also choose which quality they want to have for the meshing.

### URL of deployed dev instance (used for testing):
- https://addsegmentslist.webknossos.xyz

### Steps to test:
- create a new annotation (with or without fallback layer) and brush --> a segment should appear in the segments tab
- open a dataset in view mode and hover existing segments --> the segments should appear, too
- open a dataset in view mode and compute meshes (precomputed or adhoc) --> the segments should appear, too
- monkey-test everything around segmentations, meshes and mappings
- please also have an eye on performance (for example, try to have more than 100 segments by shift+clicking segments in an annotation with fallback layer). overall wk performance should not be slowed down. the segment list does not use virtual rendering, but shouldn't cost anything, if the list is hidden. so, we can do further perf tuning as a follow-up.

### Issues:
- fixes #4013
- fixes #4667
- fixes #5711

------
